lib/tasks/book2.rake

```diff
@@ -74,140 +74,147 @@ task :remote_genbook2 => :environment do
-          xlink.each do |link|
-            xref = link.first
-            html.gsub!(/href=\"1-.*?\.html\##{xref}\"/, "href=\"ch00/#{xref}\"") rescue nil
+          if xlink = html.scan(/href=\"1-.*?\.html\#(.*?)\"/)
+            xlink.each do |link|
+              xref = link.first
+              html.gsub!(/href=\"1-.*?\.html\##{xref}\"/, "href=\"ch00/#{xref}\"") rescue nil
+            end
           end
-        end
 
-        if xlink = html.scan(/href=\"\#(.*?)\"/)
-          xlink.each do |link|
-            xref = link.first
-            html.gsub!(/href=\"\##{xref}\"/, "href=\"ch00/#{xref}\"") rescue nil
+          if xlink = html.scan(/href=\"\#(.*?)\"/)
+            xlink.each do |link|
+              xref = link.first
+              html.gsub!(/href=\"\##{xref}\"/, "href=\"ch00/#{xref}\"") rescue nil
+            end
           end
-        end
 
-        if subsec = html.scan(/<img src="(.*?)"/)
-          subsec.each do |sub|
-            sub = sub.first
-            html.gsub!(/<img src="#{sub}"/, "<img src=\"/book/en/v2/#{sub}\"") rescue nil
+          if subsec = html.scan(/<img src="(.*?)"/)
+            subsec.each do |sub|
+              sub = sub.first
+              html.gsub!(/<img src="#{sub}"/, "<img src=\"/book/en/v2/#{sub}\"") rescue nil
+            end
           end
-        end
-
-        puts "\t\t#{chapter_type} #{chapter_number}.#{section} : #{chapter_title} . #{section_title} - #{html.size}"
 
-        csection = schapter.sections.where(:number => section).first_or_create
-        csection.title = section_title.to_s
-        csection.html = pretext + html
-        csection.save
+          puts "\t\t#{chapter_type} #{chapter_number}.#{section} : #{chapter_title} . #{section_title} - #{html.size}"
 
-        xref = Xref.where(:book_id => book.id, :name => id_xref).first_or_create
-        xref.section = csection
-        xref.save
+          csection = schapter.sections.where(:number => section).first_or_create
+          csection.title = section_title.to_s
+          csection.html = pretext + html
+          csection.save
 
-        # record all the xrefs
-        (sec.search(".//*[@id]")).each do |id|
-          id_xref = id.attribute('id').to_s
           xref = Xref.where(:book_id => book.id, :name => id_xref).first_or_create
           xref.section = csection
           xref.save
+
+          # record all the xrefs
+          (sec.search(".//*[@id]")).each do |id|
+            id_xref = id.attribute('id').to_s
+            xref = Xref.where(:book_id => book.id, :name => id_xref).first_or_create
+            xref.section = csection
+            xref.save
+          end
+
+          section += 1
+          pretext = ""
         end
+      end
 
-        section += 1
-        pretext = ""
+      book.sections.each do |section|
+        section.set_slug
+        section.save
       end
+
+    rescue Exception => msg
+      puts msg
     end
+    # consider the revision as processed whatever the result
```

Comment:
I didn't know the publishing script was run only once a day.  If it doesn't  make difference to run and fail each time, why not,

For the silent failing, the log of the exception could be appended to some part of the content, in a hidden paragraph, for instance (I don't think we can send an e-mail from the script). The problem is that coupling it to the daily run would make it more complex to manage.